sootup.core/src/main/java/sootup/core/jimple/common/ref/JArrayRef.java

```diff
@@ -84,7 +87,14 @@ public Stream<Value> getUses() {
   @Override
   @Nonnull
   public Type getType() {
-    return base.getType();
+    Type baseType = base.getType();
+    if (baseType.equals(NullType.getInstance())) {
+      return NullType.getInstance();
+    } else if (baseType instanceof ArrayType) {
+      return ((ArrayType) baseType).getElementType();
```

Comment:
`((ArrayType) baseType).getElementType();` handles the `int[ ][ ]` case I guess
https://github.com/soot-oss/SootUp/blob/621cabd9bd6dbcc898189247e5884bd8952c2003/sootup.core/src/main/java/sootup/core/types/ArrayType.java#L90-L96

Reply:
`base.getType()` is  `UnknownType` when `TypeAssigner` is not enabled.
I feel it hard to use if we throw errors inside getType. We have to do try-catch every where.

Reply:
But baseType would already be unknown, so we can return it directly / same for NullType - Im wondering if handling just ArrayType/return baseType otherwise would be enough?

Reply:
The current implementation is correct you want to have the type of a entry in an array. if the array is multidimensional you should get an array as return value.

Reply:
I get the point. 
`else return base.getType()` is equivalent from the value perspective.
But this code would imply that the return type of jarrayref is just the array itself, this could be confusing.

Reply:
The current implementation is correct you want to have the type of a entry in an array. if the array is multidimensional you should get an array as return value.

Reply:
it makes sense.. as these two would be the only valid other base types as the ArrayRef should not point anywhere else than to an Array which is of ArrayType.